nightfall-optimist/src/routes/proposer.mjs

```diff
@@ -79,13 +81,18 @@ router.post('/register', async (req, res, next) => {
         logger.warn(
           'Proposer was already registered on the blockchain but not with this Optimist instance - registering locally',
         );
-        const stateContractInstance = await waitForContract(STATE_CONTRACT_NAME);
-        const currentProposer = await stateContractInstance.methods.getCurrentProposer().call();
         if (address === currentProposer.thisAddress) {
           proposer.isMe = true;
           await enqueueEvent(() => logger.info('Start Queue'), 0); // kickstart the queue
         }
       }
+    } else if (address === currentProposer.thisAddress && !proposer.isMe) {
+      logger.warn(
+        'Proposer was already registered on the blockchain and with this Optimist instance, but proposer flag wasnt set - setting isMe flag',
+      );
+      proposer.isMe = true;
+      proposer.address = address;
+      await enqueueEvent(() => logger.info('Start Queue'), 0); // kickstart the queue
```

Comment:
Can you comment on this?
What is the situation we are trying to resolve :pray: Thank you

Reply:
there is a scenario that i found in testing that this proposer's address was the current proposer's address, but the flag `proposer.isMe` wasn't set. It only happened at boot time from time to time.... so i just forced it to be the same